q2_alignment/_sina.py

```diff
@@ -0,0 +1,88 @@
+# ----------------------------------------------------------------------------
+# Copyright (c) 2018, QIIME 2 development team.
+#
+# Distributed under the terms of the Modified BSD License.
+#
+# The full license is in the file LICENSE, distributed with this software.
+# ----------------------------------------------------------------------------
+
+__author__ = "Elmar Pruesse"
+
+import os.path as op
+import subprocess as sp
+from tempfile import TemporaryDirectory
+
+from q2_types.feature_data import AlignedDNAFASTAFormat, DNAFASTAFormat
+from qiime2.plugin import Int, Str
+
+import skbio
+import skbio.io
+
+
+def _run_command(cmd):
+    print("# Running command: " + " ".join(cmd))
+    sp.run(cmd, check=True)
+    print("# Command "+cmd[0]+" finished")
+
+
+def check_no_duplicate_fasta_id(fasta_fp, msg="reference"):
```

Comment:
This check could be slow for big input files - does sina not perform this check on its own?

Reply:
No, SINA allows this. When writing to FASTA, the duplicate IDs don't matter (to SINA). When writing to an ARB database, overwriting a sequence already present in the database may be intended (e.g. when aligning from/to the same databases).

Technically, I suppose it'd be best if `AlignedDNAFASTAFormat` could guarantee that the IDs are unique - it's not really valid/sensible multi-FASTA if you've got duplicate IDs in there.

Reply:
Re time: the indexing will be slower at this time, so shouldn't be much of an issue.

Reply:
:+1: I added an issue for this here: https://github.com/qiime2/q2-types/issues/198